crates/wysiwyg/src/dom/nodes/container_node.rs

```diff
@@ -334,6 +359,108 @@ where
             self.append_child(child);
         }
     }
+
+    /// Slice this container after given position.
+    /// Returns a new container of the same kind with the
+    /// removed content, with both nodes keeping
+    /// expected hierarchy.
+    pub fn slice_after(&mut self, offset: usize) -> ContainerNode<S> {
```

Comment:
And also a very similar function to this one, to split the Dom into a sub-tree that can be inserted later. I guess we'll have to try merge them somehow later.

Also, could we change `offset` for `index`? Offset is usually used in the project for char offset and it can be a bit misleading.

Reply:
I'm changing for `position` because this is actually a relative position inside the node more than an index or an offset, actually even my documentation says so 😅